pkg/tsdb/cloudwatch/time_series_query_test.go

```diff
@@ -136,3 +136,224 @@ func TestTimeSeriesQuery(t *testing.T) {
+	MetricQueryType  metricQueryType  `json:"metricQueryType"`
+	MetricEditorMode metricEditorMode `json:"metricEditorMode"`
+	Dimensions       queryDimensions  `json:"dimensions"`
+	Expression       string           `json:"expression"`
+	Alias            string           `json:"alias"`
+	Statistic        string           `json:"statistic"`
+	Period           string           `json:"period"`
+	MatchExact       bool             `json:"matchExact"`
+	MetricName       string           `json:"metricName"`
+}
+
+func newTestQuery(t testing.TB, p queryParameters) json.RawMessage {
+	t.Helper()
+
+	tsq := struct {
+		Type             string           `json:"type"`
+		MetricQueryType  metricQueryType  `json:"metricQueryType"`
+		MetricEditorMode metricEditorMode `json:"metricEditorMode"`
+		Namespace        string           `json:"namespace"`
+		MetricName       string           `json:"metricName"`
+		Dimensions       struct {
+			InstanceID []string `json:"InstanceId,omitempty"`
+		} `json:"dimensions"`
+		Expression string `json:"expression"`
+		Region     string `json:"region"`
+		ID         string `json:"id"`
+		Alias      string `json:"alias"`
+		Statistic  string `json:"statistic"`
+		Period     string `json:"period"`
+		MatchExact bool   `json:"matchExact"`
+		RefID      string `json:"refId"`
+	}{
+		Type:   "timeSeriesQuery",
+		Region: "us-east-2",
+		ID:     "query id",
+		RefID:  "A",
+
+		MatchExact:       p.MatchExact,
+		MetricQueryType:  p.MetricQueryType,
+		MetricEditorMode: p.MetricEditorMode,
+		Dimensions:       p.Dimensions,
+		Expression:       p.Expression,
+		Alias:            p.Alias,
+		Statistic:        p.Statistic,
+		Period:           p.Period,
+		MetricName:       p.MetricName,
+	}
+
+	marshalled, err := json.Marshal(tsq)
+	require.NoError(t, err)
+
+	return marshalled
+}
+
+func Test_QueryData_response_data_frame_names(t *testing.T) {
+	origNewCWClient := NewCWClient
+	t.Cleanup(func() {
+		NewCWClient = origNewCWClient
+	})
+	var cwClient fakeCWClient
+	NewCWClient = func(sess *session.Session) cloudwatchiface.CloudWatchAPI {
+		return &cwClient
+	}
+
+	cwClient = fakeCWClient{
+		GetMetricDataOutput: cloudwatch.GetMetricDataOutput{
+			MetricDataResults: []*cloudwatch.MetricDataResult{
+				{StatusCode: aws.String("Complete"), Id: aws.String("query id"), Label: aws.String("response label"),
+					Values: []*float64{aws.Float64(1.0)}, Timestamps: []*time.Time{{}}},
+			},
+		},
+	}
+	im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		return datasourceInfo{}, nil
+	})
+	executor := newExecutor(im, newTestConfig(), &fakeSessionCache{})
+
+	t.Run("where user defines search expression and alias is defined, then frame name prioritizes period and stat from expression over input", func(t *testing.T) {
+		query := newTestQuery(t, queryParameters{
+			MetricEditorMode: MetricEditorModeRaw, // contributes to isUserDefinedSearchExpression = true
```

Comment:
Even though it's inferred, it would be good to set `MetricQueryType: MetricQueryTypeSearch` since that's also a prerequisite for this to be considered a isUserDefinedSearchExpression.